src/Http/Http.Extensions/gen/StaticRouteHandlerModel/StaticRouteHandlerModel.Emitter.cs

```diff
@@ -180,7 +180,9 @@ public static void EmitFilteredInvocation(this Endpoint endpoint, CodeWriter cod
     {
         if (endpoint.Response?.IsVoid == true)
         {
-            codeWriter.WriteLine($"handler({endpoint.EmitFilteredArgumentList()});");
+            codeWriter.WriteLine(endpoint.Response?.IsAwaitable == true
+                ? $"await handler({endpoint.EmitFilteredArgumentList()});"
```

Comment:
Should we be doing the completed check? Or are we already in an async state machine?
```
var task = handler(...);
if (task.IsCompleted)
{
    // ...
}
else
{
    Await(task);
}

async Task Await(Task task)
{
    // ...
}
```

Reply:
We're already in a async handler here. Unlike the RDF, the RDG is able to generate different async code so we don't need to have these completed checks in the here, I think.